### ISSUE TITLE
[BUG] Azure.Messaging.EventHubs.Processor seems to require access rights to the event hub itself (does not work with access to consumer group)

### ISSUE THREAD

**Issue by ilya-scale** [Azure/azure-sdk-for-net#45372]
### Library name and version

Azure.Messaging.EventHubs.Processor 5.11.4

### Describe the bug

I have configured Entra ID access rights to this scope:

eventhubs/{evh-name}/consumergroups/{consumer-group-name}" 

I have inherited from `PluggableCheckpointStoreEventProcessor<EventProcessorPartition>` and run `StartProcessingAsync`

But what I do get (unless I change the scope to the eventhubs (i.e. remove the consumergroups part) is an exception:

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Attempted to perform an unauthorized operation.
For troubleshooting information, see https://aka.ms/azsdk/net/eventhubs/exceptions/troubleshoot)
 ---> System.UnauthorizedAccessException: Attempted to perform an unauthorized operation.
For troubleshooting information, see https://aka.ms/azsdk/net/eventhubs/exceptions/troubleshoot
   at Azure.Messaging.EventHubs.AmqpError.ThrowIfErrorResponse(AmqpMessage response, String eventHubName)
   at Azure.Messaging.EventHubs.Amqp.AmqpClient.GetPropertiesAsync(EventHubsRetryPolicy retryPolicy, CancellationToken cancellationToken)
   at Azure.Messaging.EventHubs.Amqp.AmqpClient.GetPropertiesAsync(EventHubsRetryPolicy retryPolicy, CancellationToken cancellationToken)
   at Azure.Messaging.EventHubs.EventHubConnection.GetPropertiesAsync(EventHubsRetryPolicy retryPolicy, CancellationToken cancellationToken)
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.ValidateEventHubsConnectionAsync(CancellationToken cancellationToken)
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.ValidateEventHubsConnectionAsync(CancellationToken cancellationToken)
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.ValidateProcessingPreconditions(CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.StartProcessingInternalAsync(Boolean async, CancellationToken cancellationToken)
   at Azure.Messaging.EventHubs.Primitives.EventProcessor`1.StartProcessingAsync(CancellationToken cancellationToken)
   ...
```

After looking at the code it seems that there is a validation that the connection is valid and that validation does not use consumer group, but goes right to the event hub, thus requiring higher level of access rights.

P.S. I did not find any sufficient documentation on how to do that properly though so that might be an incorrect scope, so if you can point me to the right docs that would be nice, but I suppose it just does not exist :(

I also did not find any documentation on the library that says this is not supported

### Expected behavior

It should work without an error

### Actual behavior

It fails as described above

### Reproduction Steps

Use the package to connect with a service principal that has access only to consumer group

**Comment by jsquire** [comment]
Hi @ilya-scale.   Your observation is correct, and this is by design.  The processor requires metadata (such as what partitions exist) which are exposed by the Event Hub management operations and not associated with a specific consumer group.

**Comment by ilya-scale** [comment]
/unresolve Thanks for the explanation, it would be good to reflect this somewhere in the documentation perhaps?

**Comment by ilya-scale** [comment]
Thanks for the explanation, it would be good to reflect this somewhere in the documentation perhaps and keep this issue for that?

**Comment by jsquire** [comment]
fixed by #45499

**Comment by jsquire** [comment]
The requested docs will go out as part of tomorrow's release.
